Source/CsDebugScript.Engine/UserType.cs

```diff
@@ -17,12 +17,26 @@ public static class VariableCastExtender
         /// <param name="variable">The variable.</param>
         public static T DynamicCastAs<T>(this Variable variable) where T : UserType
         {
-            if (variable == null || !variable.GetRuntimeType().Inherits<T>())
+            if (variable == null)
             {
                 return null;
             }
 
-            return variable.CastAs<T>();
+            CodeType runtimeType = variable.DereferencePointer().GetRuntimeType();
+
+            if (runtimeType.IsPointer)
+            {
+                runtimeType = runtimeType.ElementType;
+            }
+
+            if (runtimeType.Inherits<T>())
+            {
+                return variable.CastAs<T>();
```

Comment:
This one won't work with multi inheritance. You should do DowncastInterface to upper type and call Variable.GetBaseClass to get correct variable...